왜 Coverage가 안되는걸까요...

icarus3023 opened the pull request.

icarus3023 committed:
test

icarus3023 committed:
test

icarus3023 committed:
Merge branch 'master' of https://github.com/Acloy/real.git

icarus3023 committed:
Input jacoco

icarus3023 committed:
target source change to 1.7

tony-riot commented:
Maven goal로  package를 주고 실행시키신거 아닌가요? PPT에 보시면 Coverage를 위해 돌려야 할 goal은 package가 아닌 다른 것을 지정하라고 되어 있습니다. 해당 goal로 돌려보세요.

icarus3023 commented:
jenkins에서 Goal을 clean org.jacoco:jacoco-maven-plugin:prepare-agent install으로 바꾸었는데도 나오지 않네요

tony-riot commented:
Jenkins workspace build 로그를 보내주시면 한번 더 확인해보시는게 좋을 것 같습니다.
빌드 과정에서 이상이 없다면 현재 설정으로 봐서는 정상적으로 데이터를 볼 수 있어야 할 것 같네요.

icarus3023 committed:
빌드기록 첨부

icarus3023 commented:
JaCoCoItSensor: JaCoCo IT report not found: C:\Jenkins\jobs\real\workspace\target\jacoco-it.exec
빌드 로그 확인해보니 이러한 내용이 있던데요... jacooco가 설치 안되어서 그런걸까요?
빌드기록

tony-riot commented:
pom.xml 파일의 플러그인 설정 부분에서 다음 부분이 누락되었네요.

```
    <executions>
      <execution>
        <id>default-prepare-agent</id>
        <goals>
          <goal>prepare-agent</goal>
        </goals>
      </execution>
    </executions>
```

이 부분을 추가한 다음에 다시 한번 해보세요.

icarus3023 committed:
<executions>
      <execution>
        <id>default-prepare-agent</id>
        <goals>
          <goal>prepare-agent</goal>
        </goals>
      </execution>
    </executions>
    첨부

icarus3023 commented:
추가후 메이븐으로 빌드하고 git 커밋&푸쉬 젠킨스 빌드해야하는건가요?

icarus3023 commented:
Jenkins에서 Analysis properties에 대하여 궁금한점이 있습니다.
sonar.projectKey는 pom.xml에서 <groubId>:<artifactId> sonar.projectName은 pom.xml에서의 <name>으로 해야 하는거 아닌가요?

tony-riot commented:
정확하게 어떤 질문을 하시는지 모르겠네요.
pom.xml 에서 어떤 걸 해야한다는건지?

On Wed, Mar 30, 2016 at 11:13 PM, icarus3023 notifications@github.com
wrote:

> Jenkins에서 Analysis properties에 대하여 궁금한점이 있습니다.
> sonar.projectKey는 pom.xml에서 : sonar.projectName은 pom.xml에서의 으로 해야 하는거 아닌가요?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> https://github.com/tony-riot/real/pull/2#issuecomment-203451016

## 

지두현 / Tony Chi
Software Engineer
tchi@riotgames.com
+82-10-4143-2031

icarus3023 commented:
Jenkins에서 Post Steps의 Invoke Standalone SonarQube Analysis 안에서
Analysis properties에서 sonar.projectKey는 어떤 걸 해야하는건가요?
ppt에서는 workspace에 맞게 해줘야한다고 하는데요 sonar runner doc에서는 maven을 사용할때는 projectKey가 <groubID>:<artifactID>로 하라고 나와있더라구요.

tony-riot commented:
http://docs.sonarqube.org/display/SONAR/Analysis+Parameters 설명에 보시면...

sonar.projectKey 는 그냥 Unique한 값을 잡으라고 하네요. 그래서 보통 groupId 이름과 artifactId를 :
를 사이에 구분자로 놓고 합쳐서 사용한다고 설명이 나와있습니다.
":" 를 사용하라는게 아니고... -_-;;

참고하세요.

On Wed, Mar 30, 2016 at 11:25 PM, icarus3023 notifications@github.com
wrote:

> Jenkins에서 Post Steps의 Invoke Standalone SonarQube Analysis 안에서
> Analysis properties에서 sonar.projectKey는 어떤 걸 해야하는건가요?
> ppt에서는 workspace에 맞게 해줘야한다고 하는데요 sonar runner doc에서는 maven을 사용할때는
> projectKey가 :로 하라고 나와있더라구요.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> https://github.com/tony-riot/real/pull/2#issuecomment-203456496

## 

지두현 / Tony Chi
Software Engineer
tchi@riotgames.com
+82-10-4143-2031

icarus3023 commented:
pom에서 
<executions>
      <execution>
        <id>default-prepare-agent</id>
        <goals>
          <goal>prepare-agent</goal>
        </goals>
      </execution>
    </executions>
추가하였는데도 
INFO: JaCoCoSensor: JaCoCo report not found : C:\Jenkins\jobs\helloworld\workspace\target\jacoco.exec
INFO: JaCoCoItSensor: JaCoCo IT report not found: C:\Jenkins\jobs\helloworld\workspace\target\jacoco-it.exec
INFO: JaCoCoOverallSensor: JaCoCo reports not found.
이러한 에러가 나옵니다..

tony-riot commented:
​pom.xml 파일에서 java source를 1.7에서 1.8로 변경해보세요.
이후에도 안된다면 전체 로그를 한번 더 보내주세요.

On Thu, Mar 31, 2016 at 7:26 AM, icarus3023 notifications@github.com
wrote:

> pom에서
> 
> default-prepare-agent
> 
> prepare-agent
> 
> 추가하였는데도
> INFO: JaCoCoSensor: JaCoCo report not found :
> C:\Jenkins\jobs\helloworld\workspace\target\jacoco.exec
> INFO: JaCoCoItSensor: JaCoCo IT report not found:
> C:\Jenkins\jobs\helloworld\workspace\target\jacoco-it.exec
> INFO: JaCoCoOverallSensor: JaCoCo reports not found.
> 이러한 에러가 나옵니다..
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> https://github.com/tony-riot/real/pull/2#issuecomment-203665906

## 

지두현 / Tony Chi
Software Engineer
tchi@riotgames.com
+82-10-4143-2031

icarus3023 committed:
sorce&target 1.8로 변경

icarus3023 committed:
target 1.7변경

icarus3023 committed:
Update pom.xml

icarus3023 committed:
build_22_changesource

icarus3023 committed:
build_23_changesourcetarget

icarus3023 commented:
변경하면 이러한 에러가 나옵니다. 
실행은되나 coverage가안됩니다.
